kairon/async_callback/utils.py

```
@@ -491,3 +495,17 @@ def return_response(data: any, message : str, error_code: int, response_type: st
+    @staticmethod
+    def save_as_pdf(text: str, filename: str, bot: str, sender_id:str):
+        try:
+            _, media_id = asyncio.run(UserMedia.save_markdown_as_pdf(
+                bot=bot,
+                sender_id=sender_id,
+                text=text,
+                filepath=filename
+            ))
+            return media_id
+        except Exception as e:
+            raise Exception(f"encryption failed-{str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)